File: TopInterview150/TwoPointers/ValidPalindrome/solution.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int p1 = 0;
        int p2 = s.length()-1;
        cout << p1 << ", " << p2;
        while (p1 < p2)
        {
            if (!isalnum(s[p1]))
            {
                p1++;
            }
            else if (!isalnum(s[p2]))
            {
                p2--;
            }
            else if (tolower(s[p1]) == tolower(s[p2]))
            {
                p1++;
                p2--;
            }
            else return false;
        }
        return true;
    
    }
};

File: TopInterview150/TwoPointers/ValidPalindrome/solution.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        p1 = 0
        p2 = len(s)-1
        while (p1 < p2):
            if not s[p1].isalnum():
                p1+=1
            elif not s[p2].isalnum():
                p2-=1
            elif s[p1].lower() == s[p2].lower():
                p1+=1
                p2-=1
            else:
                return False
        return True